storage/src/test/java/org/apache/kafka/server/log/remote/storage/RemoteTopicPartitionDirectory.java

```diff
@@ -0,0 +1,169 @@
+        listFilesets().stream().forEach(fileset -> traverser.visitSegment(fileset));
+    }
+
+    private List<RemoteLogSegmentFileset> listFilesets() {
+        Set<Uuid> uuids = Arrays.stream(directory.listFiles())
+                .map(file -> getUuid(file.getName()))
+                .collect(toSet());
+
+        return uuids.stream()
+                .map(uuid -> RemoteLogSegmentFileset.openExistingFileset(this, uuid))
+                .collect(Collectors.toList());
+    }
+
+    /**
+     * Creates a new {@link RemoteTopicPartitionDirectory} instance for the directory of the
+     * provided topicPartition under the root directory of the local tiered storage.
+     */
+    public static RemoteTopicPartitionDirectory openTopicPartitionDirectory(final TopicIdPartition topicIdPartition,
+                                                                            final File storageDirectory) {
+
+
+        final File directory = new File(storageDirectory, toString(topicIdPartition));
+        final boolean existed = directory.exists();
+
+        if (!existed) {
+            LOGGER.info("Creating directory: " + directory.getAbsolutePath());
+            directory.mkdirs();
+        }
+
+        return new RemoteTopicPartitionDirectory(topicIdPartition, directory, existed);
+    }
+
+    /**
+     * Creates a new {@link RemoteTopicPartitionDirectory} instance for the directory with the given
+     * name under the root directory of the local tiered storage. This method throws an
+     * {@link IllegalArgumentException} if the directory does not exist.
+     */
+    public static RemoteTopicPartitionDirectory openExistingTopicPartitionDirectory(final String dirname,
+                                                                                    final File storageDirectory) {
+        final Uuid uuid = Uuid.fromString(substr(dirname, GROUP_UUID));
+        final String topic = substr(dirname, GROUP_TOPIC);
+        final int partition;
+        try {
+            partition = Integer.parseInt(substr(dirname, GROUP_PARTITION));
+        } catch (NumberFormatException ex) {
+            throw new IllegalArgumentException(format(
+                    "Invalid format for topic-partition directory: %s", dirname), ex);
+        }
+
+        TopicPartition tp = new TopicPartition(topic, partition);
+        TopicIdPartition idPartition = new TopicIdPartition(uuid, tp);
+        final RemoteTopicPartitionDirectory directory = openTopicPartitionDirectory(idPartition, storageDirectory);
+        if (!directory.didExist()) {
+            throw new IllegalArgumentException(format("Topic-partition directory %s not found", dirname));
+        }
+        return directory;
+    }
+
+    private static String toString(TopicIdPartition topicIdPartition) {
+        Uuid uuid = topicIdPartition.topicId();
+        TopicPartition tp = topicIdPartition.topicPartition();
+        return uuid.toString() + "-" + tp.partition() + "-" + tp.topic();
```

Comment:
In case we want to adopt the new structure:
```suggestion
        return tp.topic() + "-" + tp.partition() + "-" + uuid.toString();
```